site/public/js/gradeable-notebook.js

```diff
@@ -94,43 +130,54 @@ function saveNotebookToLocal() {
-                $(element).prop('checked', values.includes(element.value)).change();
+        //to prevent data loss when these changes get installed on production
+        const not_found = [];
+
+        //restore multiple choice
+        for (const id in inputs.multiple_choice) {
+            const filename = inputs.multiple_choice[id][0];
+            const answers = inputs.multiple_choice[id][1];
+            let found = false;
+            $(`fieldset.mc[data-filename="${filename}"] input`).each(function() {
+                found = true;
+                //check off proper inputs
+                for (let i = 0; i < answers.length; i++) {
+                    if ($(this)[0].defaultValue === answers[i]) {
+                        $(this).prop('checked', true);
+                    }
+                }
             });
+            if (!found) {
+                not_found.push(answers);
+            }
         }
-        // Next, we restore short-answer boxes
-        for (const id in state.short_answer) {
-            const answer = state.short_answer[id][0];
-            // Restore the answer and trigger change events (for setting button states)
-            // (see https://stackoverflow.com/questions/4672505/why-does-the-jquery-change-event-not-trigger-when-i-set-the-value-of-a-select-us)
-            $(`#${id}`).val(answer).trigger('input');
+
+        //restore short answers
+        for (const id in inputs.short_answer) {
+            const filename = inputs.short_answer[id][0];
+            const answers = inputs.short_answer[id][1];
+            let found = false;
+            $(`.short_answer > div[data-filename="${filename}"]`).each(function(){
+                //set input
+                found = true;
+                const editor = ($(this)[0]).querySelector('.CodeMirror').CodeMirror;
+                editor.setValue(answers);
+            });
+            if (!found) {
+                not_found.push(answers);
+            }
         }
-        // Finally, we restore codeboxes
-        for (const id in state.codebox) {
-            const answer = state.codebox[id][0];
-            const codebox = $(`#${id} .CodeMirror`).get(0);
-            // If this box no longer exists, then don't attempt to update the
-            // answer. The autosave data is probably for an older version of
-            // the gradeable at this point, see issue #5351.
-            if (!codebox) {
-                continue;
+
+        if (not_found.length > 0) {
+            for (const id in not_found) {
+                console.log(not_found[id]);
             }
-            const cm = codebox.CodeMirror;
-            // This automatically triggers the event handler for the clear and
-            // recent buttons.
-            cm.setValue(answer);
+            alert(`Answer(s) could not be restored you will have to copy and paste them in the proper place, the answers are in your console right now, press f12 to open it. Answers: ${not_found}`);
```

Comment:
The instructions for how to open the console vary greatly between browsers, so I'm not sure if it is the best spot to put for users, especially because it can be very intimidating if they have never seen/used it before. Perhaps you could just create an element on the page and just insert the text blob there? It doesn't have to be pretty. Maybe try something like 
```
//create container to hold old answers
const old_answers_div = document.createElement('div');
old_answers_div.id = 'old-answers';
old_answers_div.classList.add ('box ', 'red-background');

//create header text to warn user
const old_answers_header = document.createElement('h4');
old_answers_header.innerHTML = "Answer(s) could not be restored you will have to copy and paste them in the proper place";
old_answers_header.classList.add('red-message');
//add header to container
old_answers_div.appendChild(old_answers_header);

// for ... of loop loops through all elements in array, while for ... in loop loops through
// all indexes/keys in arrays/objects
for (const answer of not_found) {
    const answer_text = document.createElement('p');
    answer_text.innerHTML = answer;
    old_answers_div.appendChild(answer_text);
}
$(old_answers_div).insertAfter('#gradeable-info');
```
This would put the answers in a div at the top of the page just below the gradeable info. Other styling may be necessary